packages/amount-selectors/src/components/AmountSelectors.js

```diff
@@ -0,0 +1,125 @@
+import * as React from "react";
+import PropTypes from "prop-types";
+import Button, { buttonTypes, buttonSizes } from "@crave/farmblocks-button";
+import InputText from "@crave/farmblocks-input-text";
+
+import Wrapper from "../styledComponents/AmountSelector";
+
+class AmountSelectors extends React.Component {
+  state = {
+    value: this.props.value,
+    validationMessages: []
+  };
+
+  isDigit = value => {
+    return !Number.isNaN(parseFloat(value));
+  };
+
+  isTypedKeyValid = key => {
+    const commonAcceptedKeys = [
+      "Backspace",
+      "ArrowRight",
+      "ArrowLeft",
+      "Delete",
+      "."
+    ];
+
+    const isValidKey = commonAcceptedKeys.indexOf(key) !== -1;
+
+    return this.isDigit(key) || isValidKey;
+  };
+
+  onKeyDown = event => {
+    if (!this.isTypedKeyValid(event.key)) {
+      event.stopPropagation();
+      event.preventDefault();
+    }
+  };
+
+  onChange = event => {
+    const value = typeof event === "number" ? event : event.target.value;
+    const parsedValue = parseFloat(value);
+
+    if (
+      this.isDigit(parsedValue) &&
+      parsedValue >= this.props.min &&
+      parsedValue <= this.props.max
+    ) {
+      this.setState({
+        value: parsedValue,
+        validationMessages: []
+      });
+
+      return this.props.onChange(parsedValue);
+    }
+
+    this.setState({
+      validationMessages: ["Invalid value"]
```

Comment:
> Type something
> Clear everything


![image](https://user-images.githubusercontent.com/17936244/40366372-56af6ae0-5dad-11e8-8210-16ab102fc73f.png)

Reply:
Actually we can remove the `validationMessages`